src/main/groovy/net/wasdev/wlp/gradle/plugins/tasks/InstallAppsTask.groovy

```diff
@@ -137,24 +145,126 @@ class InstallAppsTask extends AbstractServerTask {
+        looseWar.addSourceDir()
+        looseWar.addOutputDir(looseWar.getDocumentRoot(), task, "/WEB-INF/classes");
+
+        //retrieves dependent library jar files
+        addWarEmbeddedLib(looseWar.getDocumentRoot(), looseWar, task);
+
+        //add Manifest file
+        File manifestFile = new File(project.sourceSets.main.getOutput().getResourcesDir().getParentFile().getAbsolutePath() + "/META-INF/MANIFEST.MF")
+        looseWar.addManifestFile(manifestFile, "gradle-war-plugin")
+    }
+
+    private void addWarEmbeddedLib(Element parent, LooseApplication looseApp, Task task) throws Exception {
+      ArrayList<File> deps = new ArrayList<File>();
+      project.configurations.runtime.each { deps.add(it)}
+      project.configurations.providedRuntime.each { deps.remove(it) }
+      File parentProjectDir = new File(task.getProject().getRootProject().rootDir.getAbsolutePath())
+      for (File dep: deps) {
+        String projectPath = getProjectPath(parentProjectDir, dep)
+        if(!projectPath.isEmpty() && project.getRootProject().findProject(projectPath) != null){
+            Element archive = looseApp.addArchive(parent, "/WEB-INF/lib/"+ dep.getName());
+            looseApp.addOutputDirectory(archive, project.getRootProject().findProject(projectPath), "/");
+            looseApp.addManifestFile(archive, project.getRootProject().findProject(projectPath), "gradle-jar-plugin");
+        } else{
+            looseApp.getConfig().addFile(parent, dep.getAbsolutePath() , "/WEB-INF/lib/" + dep.getName());
         }
-        else {
-            throw new GradleException("Archive path not found. Supported formats are jar, war, and ear.")
+      }
+    }
+
+    private String getProjectPath(File parentProjectDir, File dep){
+      String dependencyPathPortion = dep.getAbsolutePath().replace(parentProjectDir.getAbsolutePath()+"/","")
+      String projectPath = dep.getAbsolutePath().replace(dependencyPathPortion,"")
+      Pattern pattern = Pattern.compile("/build/.*")
+      Matcher matcher = pattern.matcher(dependencyPathPortion)
+      projectPath = matcher.replaceAll("")
+      return projectPath;
+    }
+
+    private boolean isSupportedType(){
+      switch (getPackagingType()) {
+        case "ear":
+        case "war":
+            return true;
+        default:
+            return false;
         }
     }
+    private String getLooseConfigFileName(Task task){
+      return getArchiveName(task) + ".xml"
+    }
+
+    private String getPackagingType() throws Exception{
+      if (project.plugins.hasPlugin("ear") || !project.tasks.withType(Ear).isEmpty()) {
+          return "ear"
+      }
+      else if (project.plugins.hasPlugin("war") || !project.tasks.withType(War).isEmpty()) {
+          return "war"
+      }
+      else {
+          throw new GradleException("Archive path not found. Supported formats are jar, war, and ear.")
+      }
+  }
+
+  protected void deleteApplication(File parent, File artifactFile) throws IOException {
```

Comment:
Can you add a comment to this method saying that it cleans up the application if the install style is switched from loose to archive or vice versa.  At least, that's what I think it is doing.